src/edge-agent/Agent.Backup.ts

```diff
@@ -1,57 +1,94 @@
   ) {}
 
   /**
-   * create JWE of data stored in Pluto
+   * Creates a JWE (JSON Web Encryption) containing the backup data stored in Pluto.
+   * The data can optionally be encrypted using a custom master key and can be compressed.
+   * 
+   * @param {BackupOptions} [options] - Optional settings for the backup.
+   * @param {Version} [options.version] - Specifies the version of the backup data.
+   * @param {MasterKey} [options.key] - Custom master key used for encrypting the backup.
+   * @param {boolean} [options.compress] - If true, compresses the JWE using DEFLATE.
+   * 
+   * @returns {Promise<string>} - A promise that resolves to the JWE string.
    * 
-   * @returns {string}
-   * @see restore
+   * @see restore - Method to restore data from a JWE string.
    */
-  async createJWE(): Promise<string> {
+  async createJWE(options?: BackupOptions): Promise<string> {
     await this.Agent.pollux.start();
-    const backup = await this.Agent.pluto.backup();
-    const masterSk = await this.masterSk();
+    
+    const backup = await this.Agent.pluto.backup(options?.version);
+    const backupStr = options?.compress ? this.compress(JSON.stringify(backup)) : JSON.stringify(backup);
+    const masterSk = options?.key ?? await this.masterSk();
     const jwk = masterSk.to.JWK();
-    const jwe = this.Agent.pollux.jwe.JWE.encrypt(
-      JSON.stringify(backup),
+    
+    return this.Agent.pollux.jwe.JWE.encrypt(
+      backupStr,
       JSON.stringify(jwk),
-      'backup'
+      'backup',
     );
-    return jwe;
   }
 
   /**
-   * decode JWE and save data to store
+   * Decodes a JWE (JSON Web Encryption) string and restores the backup data to the store.
+   * If the JWE is compressed (Base64-encoded), it will attempt to decompress it first.
+   * 
+   * @param {string} jwe - The JWE string containing the encrypted backup data.
+   * @param {MasterKey} [key] - Optional custom master key used for decrypting the JWE. 
+   *                            If not provided, the default master key is used.
+   * 
+   * @returns {Promise<void>} - A promise that resolves when the data is successfully restored.
    * 
-   * @param jwe 
-   * @see backup
+   * @see createJWE - Method to create a JWE from the stored backup data.
    */
-  async restore(jwe: string) {
+  async restore(jwe: string, key?: MasterKey) {
     await this.Agent.pollux.start();
-    const masterSk = await this.masterSk();
+    const masterSk = key ?? await this.masterSk();
+
     const jwk = masterSk.to.JWK();
     const decoded = this.Agent.pollux.jwe.JWE.decrypt(
       jwe,
       'backup',
-      JSON.stringify(jwk)
+      JSON.stringify(jwk),
     );
-    const jsonStr = Buffer.from(decoded).toString();
-    const json = JSON.parse(jsonStr);
-    const backup = this.parseBackupJson(json);
+    let jsonStr: string;
+    try {
+      jsonStr = this.decopress(new TextDecoder().decode(decoded));
+    } catch {
+      jsonStr = Buffer.from(decoded).toString();
+    }    
```

Comment:
@curtis-h ok, I'll set optional flag for restoration, version side it may not good idea cause make this compress as option to createJWE so it would be unrelated to version and any version maybe compressed or not